source/package.lisp

```diff
@@ -1,21 +1,45 @@
-;;; -*- mode: Lisp; Syntax: Common-Lisp; -*-
-;;;
-;;; Copyright (c) 2009 by the authors.
-;;;
-;;; See LICENCE for details.
+;;;; SPDX-FileCopyrightText: hu.dwim & Atlas Engineer LLC
+;;;; SPDX-License-Identifier: Public Domain
 
 (in-package :common-lisp-user)
 
-(defpackage :hu.dwim.defclass-star
-  (:nicknames :class*)
+(defpackage :nclass
   (:use :common-lisp)
-  (:export #:defclass*
-           #:defcondition*
+  (:export #:define-class
+           #:define-condition*
            ;; transformers
            #:default-accessor-name-transformer
            #:dwim-accessor-name-transformer
            #:default-initarg-name-transformer
            #:default-slot-definition-transformer
            #:default-predicate-name-transformer
+           #:always-dashed-predicate-name-transformer
+           #:question-mark-predicate-name-transformer
+           #:default-type-inference
            #:make-name-transformer
-           #:*allowed-slot-definition-properties*))
+           #:*allowed-slot-definition-properties*)
+  (:documentation "This library offers two helper macros:
+- define-class
+- define-condition*
+
+Compared to the standard macros, they accept extra options and slot definition
+is smarter.
+
+Example:
+
+(define-class foo ()
+  ((slot1 :initarg nil)
+   (slot2 \"hello!\")
+   (unexported-slot :export nil))
+  (:export-class-name-p t)
+  (:export-accessor-names-p t)
+  (:accessor-name-transformer #'nclass:default-accessor-name-transformer))
+
+In the above, all slot accessors are automatically defined using
+`default-accessor-name-transformer'.  They are also exported together with the
+class name.
+The initarg default to the keyword version of the slot symbol, unless it's
+explicitly set to NIL.
+
+Notice that the second value of the slot definition, if not an option, is then
```

Comment:
I still have a desire to also allow a docstring as the third value. Should be easy enough to detect too.

But, obviously, that's beyond this PR.

Reply:
It's hard and paves the way for bugs...

Reply:
I mean, if we parse it well, then it should be solid enough. Pattern matching, maybe?

Reply:
If you can come up with something, by all means.  In any case, it belongs to a future pull request, this one is big enough already.